### ISSUE TITLE
e2e test OIDC discovery fails with ECDSA service account key

### ISSUE THREAD

**Issue by smira** [kubernetes/kubernetes#100674]
#### What happened:

e2e test `[sig-auth] ServiceAccounts ServiceAccountIssuerDiscovery should support OIDC discovery of service account issuer [Conformance]` fails with ECDSA Service Account key

#### What you expected to happen:

e2e test to succeed with ECDSA Service Account key

#### How to reproduce it (as minimally and precisely as possible):

Use ECDSA key for the service account key.

#### Anything else we need to know?:

Logs from the started test container:

```
2021/03/30 14:13:47 OK: Got token
2021/03/30 14:13:47 OK: got issuer https://172.20.0.1:6443
2021/03/30 14:13:47 Full, not-validated claims: 
openidmetadata.claims{Claims:jwt.Claims{Issuer:"https://172.20.0.1:6443", Subject:"system:serviceaccount:svcaccounts-4993:default", Audience:jwt.Audience{"oidc-discovery-test"}, Expiry:1617114224, NotBefore:1617113624, IssuedAt:1617113624, ID:""}, Kubernetes:openidmetadata.kubeClaims{Namespace:"svcaccounts-4993", ServiceAccount:openidmetadata.kubeName{Name:"default", UID:"d28680f5-302a-4824-95cd-46dc0fbf8018"}}}
2021/03/30 14:13:47 OK: Constructed OIDC provider for issuer https://172.20.0.1:6443
2021/03/30 14:13:47 oidc: id token signed with unsupported algorithm, expected ["RS256"] got "ES256"
```

`go-oidc` library defaults to `RS256` by default as the only supported method while `ES256` should be added explicitly: https://pkg.go.dev/github.com/coreos/go-oidc/v3/oidc#Config

Relevant code: https://github.com/kubernetes/kubernetes/blob/master/test/images/agnhost/openidmetadata/openidmetadata.go#L100

On the API server side:

```
$ curl -k --cert client.crt --key client.key  https://172.20.0.2:6443/.well-known/openid-configuration
{"issuer":"https://172.20.0.1:6443","jwks_uri":"https://172.20.0.1:6443/openid/v1/jwks","response_types_supported":["id_token"],"subject_types_supported":["public"],"id_token_signing_alg_values_supported":["ES256"]}
```

#### Environment:
- Kubernetes version (use `kubectl version`): 1.21.0-rc.0
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`): Talos
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

**Comment by smira** [comment]
Changing the service account key type to RSA fixes the test:

```
2021/03/30 14:43:05 OK: Got token
2021/03/30 14:43:05 OK: got issuer https://172.20.0.1:6443
2021/03/30 14:43:05 Full, not-validated claims: 
openidmetadata.claims{Claims:jwt.Claims{Issuer:"https://172.20.0.1:6443", Subject:"system:serviceaccount:svcaccounts-5418:default", Audience:jwt.Audience{"oidc-discovery-test"}, Expiry:1617115983, NotBefore:1617115383, IssuedAt:1617115383, ID:""}, Kubernetes:openidmetadata.kubeClaims{Namespace:"svcaccounts-5418", ServiceAccount:openidmetadata.kubeName{Name:"default", UID:"c7e830c2-9c6a-4f48-9020-87cae65ff179"}}}
2021/03/30 14:43:05 OK: Constructed OIDC provider for issuer https://172.20.0.1:6443
2021/03/30 14:43:05 OK: Validated signature on JWT
2021/03/30 14:43:05 OK: Got valid claims from token!
2021/03/30 14:43:05 Full, validated claims: 
&openidmetadata.claims{Claims:jwt.Claims{Issuer:"https://172.20.0.1:6443", Subject:"system:serviceaccount:svcaccounts-5418:default", Audience:jwt.Audience{"oidc-discovery-test"}, Expiry:1617115983, NotBefore:1617115383, IssuedAt:1617115383, ID:""}, Kubernetes:openidmetadata.kubeClaims{Namespace:"svcaccounts-5418", ServiceAccount:openidmetadata.kubeName{Name:"default", UID:"c7e830c2-9c6a-4f48-9020-87cae65ff179"}}}
```

**Comment by BenTheElder** [comment]
SIG Auth own their test cases (it's tagged in the test name), SIG Architecture own conformance. Testing owns the frameworks, CI, tooling, etc.

**Comment by liggitt** [comment]
/assign @mtaufen 
/priority critical-urgent
/milestone v1.21

because this is in the conformance suite, it must not restrict valid+supported cluster configurations

**Comment by aojea** [comment]
/reopen
the image needs to be promoted and bumped in the e2e test

**Comment by smira** [comment]
promote image PR: https://github.com/kubernetes/k8s.io/pull/1847

**Comment by smira** [comment]
I believe with #100687 merged this can be closed as all the fixes were done and backported to 1.21

**Comment by BenTheElder** [comment]
Thanks!
